add pyblish support

hannesdelbeke opened the pull request:
add pyblish wrapper allowing checks to be loaded in [pyblish](https://github.com/pyblish/pyblish-maya)
make import paths not local to prevent import issues

plugin factory generates plugin classes from the command list
and wraps the commands in a pyblish validation class

a collector for mesh paths is included to allow running of these plugins as standalone
![image](https://user-images.githubusercontent.com/3758308/150577712-22babf81-fd5b-4655-ba29-aa9332f51573.png)

future improvements:
- speed is noticably slower in pyblish compared to the  original mesh checker.

hannesdelbeke committed:
make paths not local to prevent import issues
add pyblish wrapper allowing checks to be loaded in pyblish

JakobJK commented:
Hey Hannes,
Thanks so much for creating a PR! 
Thanks for pointing out the import issues. I have fixed those in the main branch, so this PR can be focused on the pyblish support.

What kind of discrepancy are you seeing in terms of speed? Do you have some rough numbers?

hannesdelbeke commented:
i believe the speed discrepancy to be related to pyblish QML and pyblish lite
it also happens with other pyblish plugins i've worked with
since this is not related to the mesh checker checks/functions, we can ignore it in this PR.

when running the same checks. the mesh checker finishes almost instantly. whereas pyblish can take it's sweet time.
this doesn't happen when running through pyblish headless mode, so likely is caused by a bug in the pyblish GUI

hannesdelbeke committed:
Merge branch 'main' into feature/pyblish-support

hannesdelbeke committed:
remove comments
change variable name from dict to data to prevent clash with default dict variable
add comment to describe plugin creation

hannesdelbeke committed:
add description doc support
remove unused import

hannesdelbeke commented:
made a thread on the "slowness" https://github.com/pyblish/pyblish-qml/issues/371
for the QML UI it's unclear why eactly it is so slow.
for lite it's on purpose by default to make it appear more responsive, and can be adjusted with an ENV var

also added some more commites to this PR: added descriptions for each command, copied from albertos PR so english might need a bit of cleanup

but from the discussion on his PR it sounds like you want to keep this repo as basic as possible. 
so shall we close this pyblish PR? 
I can host the pyblishplugins in my fork.
